src/Native/System.Native/pal_console.cpp

```diff
@@ -35,9 +35,10 @@ extern "C" int32_t GetWindowSize(WinSize* windowSize)
 #endif
 }
 
-extern "C" int32_t IsATty(int fd)
+extern "C" int32_t IsATty(intptr_t fd)
 {
-    return isatty(fd);
+    assert(INT32_MIN <= fd && fd <= INT32_MAX);
```

Comment:
(note that we could probably rely on the platform to do this for us by collapsing all out-of-range file descriptors to `-1`)